src/linear_algebra/finite_dimensional.lean

```diff
@@ -850,6 +850,72 @@ by { unfold findim, simp [dim_top] }
 
 end top
 
+lemma eq_zero_of_findim_eq_zero [finite_dimensional K V]
+  (hV : findim K V = 0) (x : V) : x = 0 :=
+begin
+  obtain ⟨B, hB⟩ := exists_is_basis_finset K V,
+  have := (findim_eq_card_finset_basis hB).symm,
+  rw [hV, finset.card_eq_zero, finset.eq_empty_iff_forall_not_mem] at this,
+  rw ← is_basis.equiv_fun_total hB x,
+  convert finset.sum_empty,
+  rw finset.univ_eq_empty,
+  rintro ⟨b, hB⟩,
+  exact this b hB,
+end
+
+lemma subsingleton.is_basis (h : subsingleton V) : is_basis K (λ x : fin 0, (0 : V)) :=
+begin
+  refine is_basis_empty (finset.univ_eq_empty.mp rfl) _,
+  intro x,
+  rw [subsingleton.elim x 0],
+end
+
+lemma findim_eq_zero_iff_subsingleton [finite_dimensional K V] :
+  findim K V = 0 ↔ subsingleton V :=
+begin
+  split; intro h,
+  { apply subsingleton.intro,
+    intros x y,
+    rw [eq_zero_of_findim_eq_zero h x, eq_zero_of_findim_eq_zero h y] },
```

Comment:
It looks to me like there's a missing `subsingleton_of_forall_eq (x : A) (forall y, y = x) : subsingleton A` lemma, which can probably also be used to simplify `unique.to_subsingleton`

Reply:
How about ```lemma subsingleton_of_forall_eq {α} (x : α) (h : ∀ y, y = x) : subsingleton α :=
subsingleton.intro (λ a b, (h a).symm ▸ (h b).symm ▸ rfl)```
Where should this go? I can't seem to find a good place to put it (also, it should probably be in another pr right?)

Reply:
`loguc/basic.lean` or `logic/unique.lean` both look like good places to me. I'd lean towards angle brackets rather than spelling out intro

Reply:
I've created the pr #6376